### ISSUE TITLE
Military assignments

### ISSUE THREAD

**Issue by ror6ax** [Dwarf-Therapist/Dwarf-Therapist#212]
Military UI is perhaps one of the most broken aspects of DF's UI.

I'm not proposing anything specifically here, just the concept - make military assignments possible from DT. Make "get Urist from forging to training with a sword/killing goblins" a (preferably) one-click operation as opposed to magical rituals involving 15 different menus, weapon racks and archery targets.

**Comment by cvuchener** [comment]
I have two main issues with that.

1) I think it goes beyond the scope of DT. DT manages dwarves (or more generally units), not squads, not uniforms, not buildings, not item stock and production. It is already complex enough as it is.

2) There are technical difficulties: currently DT is limited to changing values, anything that involves allocation of new objects or resizing vectors is too difficult. Although that could change in the (probably distant) future.

The military screen must be the most hated in DF, and despite 1 I thought about what I could do (not in DT but rather in a separate application). My conclusion is that I am not a GUI designer and I don't think I would be able to do much better than Toady. The military system itself is complex, not only the menus. Thus it's hard to design easy-to-use GUI for managing it. Maybe someone can design it but not me.

**Comment by ror6ax** [comment]
Fair enough. I've suspected something along those lines, but still, had to try ;)
Thanks for giving us DT, for me it's the only way to play the game.

**Comment by MatiasOETamminen** [comment]
I'm lending this space quickly: There's one specific issue. If you request workers from a vassal via a mission, the incoming merchants can be assigned to squads although they shouldn't be (and cannot be in vanilla).

**Comment by OG-github** [comment]
I'm new to DT. Is there a way to sort and have groupings of dwarves by squads? I would like to group my squads in the military tab but other tabs (like labors) would be fantastic.
